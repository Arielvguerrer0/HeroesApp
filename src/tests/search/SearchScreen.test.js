import React from 'react'
import { mount } from 'Enzyme'
import { MemoryRouter, Route } from 'react-router-dom'
import { SearchScreen } from '../../components/search/SearchScreen'

describe('Pruebas en < SearchScreen />', () => {

    const wrapper = mount(
        <MemoryRouter initialEntries={['/search']}>
            <Route path="/search" component={ SearchScreen } />
        </MemoryRouter>
    )

    test('Debe de mostrarse correctamente con los valores por defecto', () => {
    
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.alert-info').text().trim() ).toBe('Search a Hero')
    
    });

    test('Debe de mostrar a batman y el input con el valor del queryString', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        )

        expect( wrapper.find('input').prop('value') ).toBe('batman')
    });

    test('Debe de mostrar un error si no se encuentra el Hero  ', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman23123']}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        )

        expect(wrapper.find('.alert-danger').text().trim() ).toBe(`There is no a hero with batman23123`)
    })
    
    test('debe de llamar el push del history', () => {
        
        const history = {
            push: jest.fn()
        }

        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman23123']}>
                <Route
                path="/search" 
                component={ () => < SearchScreen history={ history } /> } />
            </MemoryRouter>
        );
        
        wrapper.find('input').simulate('change', {
            target: {
                name: 'searchText',
                value: 'batman'
            }
        });

        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
        });

        expect( history.push ).toHaveBeenCalledWith(`?q=batman`)

    })
    
    
    
})
