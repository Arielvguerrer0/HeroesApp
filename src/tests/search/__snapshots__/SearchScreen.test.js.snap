// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en < SearchScreen /> Debe de mostrarse correctamente con los valores por defecto 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Busca un heroe"
          type="text"
          value=""
        />
        <br />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Resultados 
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a Hero
      </div>
    </div>
  </div>
</div>
`;
